docs/router/examples/full_example.rs

```diff
@@ -0,0 +1,112 @@
+#![allow(non_snake_case)]
```

Comment:
Something I have noticed which may be my error (digging into now), is when refactoring out components with inline props, one may be required to import the components props as well as the component. 

If I am not making a mistake, is it possible to not require this as it can be slighly confusing if the code does not exist explicitly. 

Current File Structure: 
```
src
  |_ main.rs
  |_ components/
        |_ navbar.rs
              |_ NavBar
```
Include NavBar in another file (e.g. main.rs here)
`use crate::components::navbar::NavBar;`

navbar.rs (just refactoring fullstack example for now to integrate some auth flow)
```rust
#[inline_props]
pub fn NavBar(cx: Scope) -> Element {
    render! {
        nav {
            ul {
                li { Link { target: Route::Home {}, "Home" } }
                li { Link { target: Route::BlogList {}, "Blog" } }
            }
        }
        Outlet {}
    }
}
```
Analyzer Error: 
```
cannot find struct, variant or union type `NavBarProps` in this scope
not found in this scoperustcClick for full compiler diagnostic
router.rs(2, 1): consider importing this struct: `use crate::components::navbar::NavBarProps;
`
```
This is easy enough to get around, but is it something to consider abstracting?

Reply:
Moving a few things may end up looking like this from current example: 

```rust
use crate::components::{
    blog::{Blog, BlogList, BlogListProps, BlogPostProps, BlogProps},
    home::{Home, HomeProps},
    navbar::{NavBar, NavBarProps},
    page_not_found::{PageNotFound, PageNotFoundProps},
};
```

Reply:
I agree if we could avoid the import it would be easier, but I don't know if it is possible. I don't see a way to get an associated type from a function. [function item types](https://doc.rust-lang.org/reference/types/function-item.html) are unnamable, and I don't think we can get the function pointer type in a way that allows us to access associated types

My attempt:
```rust

fn test(){
    <Wrapper::<app2, _> as HasProps>::Props::IS_STATIC;
}

trait HasProps{
    type Props;

    fn takes_self(&self){}
}

struct Wrapper<F: Fn(Scope<T>) -> Element, T>{
    phantom:PhantomData<(T, F)>
}

impl<T, F: Fn(Scope<T>) -> Element> HasProps for Wrapper<F, T> {
    type Props = T;
}

#[inline_props]
pub fn app2(cx: Scope) -> Element {
    todo!()
}
```

Reply:
Would a good option be just to document this then somehow? It is really not a lot of work and the analyzer covers bases well.  May be nice to say "when using inline props, you must import props with component in files" somewhere.

Reply:
In components we can get the props of a component by using a const fn. It's called `fc_to_builder`.

I imagine we could have a similar utility function?

```
/// This utility function launches the builder method so rsx! and html! macros can use the typed-builder pattern
/// to initialize a component's props.
pub fn fc_to_builder<'a, A, T: Properties + 'a>(_: fn(Scope<'a, T>) -> A) -> T::Builder {
    T::builder()
}
```